pkg/api/cluster_types_test.go

```diff
@@ -84,7 +85,79 @@ func TestGetAvailableStrimziVersions(t *testing.T) {
 	}
 }
 
-func TestGetAvailableAndReadyStrimziVersions(t *testing.T) {
+func Test_Cluster_GetLatestAvailableStrimziVersion(t *testing.T) {
+	tests := []struct {
+		name    string
+		cluster func() *Cluster
+		want    *StrimziVersion
+		wantErr bool
+	}{
+		{
+			name: "When cluster has a non empty list of available strimzi versions the latest one is returned",
+			cluster: func() *Cluster {
+				inputStrimziVersions := []StrimziVersion{
+					{Version: "v3", Ready: true},
+					{Version: "v6", Ready: false},
+					{Version: "v7", Ready: false},
+				}
+				inputStrimziVersionsJSON, err := json.Marshal(inputStrimziVersions)
+				if err != nil {
+					panic(err)
+				}
+				res := Cluster{AvailableStrimziVersions: inputStrimziVersionsJSON}
+				return &res
+			},
+			want:    &StrimziVersion{Version: "v7", Ready: false},
+			wantErr: false,
+		},
+		{
+			name: "When cluster has an empty list of available strimzi nil is returned",
+			cluster: func() *Cluster {
+				inputStrimziVersions := []StrimziVersion{}
+				inputStrimziVersionsJSON, err := json.Marshal(inputStrimziVersions)
+				if err != nil {
+					panic(err)
+				}
+				res := Cluster{AvailableStrimziVersions: inputStrimziVersionsJSON}
+				return &res
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "When cluster has a nil list of available strimzi the empty list is returned",
+			cluster: func() *Cluster {
+				res := Cluster{AvailableStrimziVersions: nil}
+				return &res
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "When cluster has an invalid JSON an error is returned",
+			cluster: func() *Cluster {
+				res := Cluster{AvailableStrimziVersions: []byte(`"keyone": valueone`)}
+				return &res
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+			res, err := tt.cluster().GetLatestAvailableStrimziVersion()
+			gotErr := err != nil
+			if gotErr != tt.wantErr {
+				t.Errorf("wantErr: %v got: %v", tt.wantErr, err)
+			}
```

Comment:
```suggestion
			Expect(err != nil).To(Equal(tt.wantErr), "wantErr: %v got: %v", tt.wantErr, err)
```

Reply:
:+1: . This was written this way because it was a copy-paste of the already existing test `Test_Cluster_GetAvailableStrimziVersions`. I will update it for both.

Reply:
updated